test/jdk/java/lang/reflect/sealed_classes/TestSecurityManagerChecks.java

```diff
@@ -0,0 +1,260 @@
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8246778
+ * @summary Test that security checks occur for getPermittedSubclasses
+ * @library /test/lib
+ * @modules java.compiler
+ * @build jdk.test.lib.compiler.CompilerUtils jdk.test.lib.compiler.ModuleInfoMaker TestSecurityManagerChecks
+ * @run main/othervm --enable-preview TestSecurityManagerChecks named
+ * @run main/othervm --enable-preview TestSecurityManagerChecks unnamed
+ */
+
+import java.io.IOException;
+import java.lang.module.Configuration;
+import java.lang.module.ModuleFinder;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import jdk.test.lib.compiler.*;
+
+public class TestSecurityManagerChecks {
+
+    private static final ClassLoader OBJECT_CL = Object.class.getClassLoader();
+
+    public static void main(String[] args) throws Throwable {
+        if ("named".equals(args[0])) {
+            runNamedModuleTest();
+        } else {
+            runUnnamedModuleTest();
+        }
+    }
+
+    private static void runNamedModuleTest() throws Throwable {
+        Path classes = compileNamedModuleTest();
+        URL testLocation = TestSecurityManagerChecks.class
+                                                    .getProtectionDomain()
+                                                    .getCodeSource()
+                                                    .getLocation();
+
+        //need to use a different ClassLoader to run the test, so that the checks are performed:
+        ClassLoader testCL = new URLClassLoader(new URL[] {testLocation}, OBJECT_CL);
+        testCL.loadClass("TestSecurityManagerChecks")
+              .getDeclaredMethod("doRunNamedModuleTest", Path.class)
+              .invoke(null, classes);
+    }
+
+    public static void doRunNamedModuleTest(Path classes) throws Throwable {
+        Configuration testConfig = ModuleLayer.boot()
+                                              .configuration()
+                                              .resolve(ModuleFinder.of(),
+                                                       ModuleFinder.of(classes),
+                                                       List.of("test"));
+        ModuleLayer testLayer = ModuleLayer.boot()
+                                           .defineModulesWithOneLoader(testConfig,
+                                                                         OBJECT_CL);
+
+        // First get hold of the target classes before we enable security
+        Class<?> sealed = testLayer.findLoader("test").loadClass("test.Base");
+
+        //try without a SecurityManager:
+        Class<?>[] subclasses = sealed.getPermittedSubclasses();
+
+        if (subclasses.length != 3) {
```

Comment:
I suggest to check against the expected list of permitted subclasses here and also the validation in the subsequent calls to `getPermittedSubclasses` with security manager enabled.   That would help the readers easier to understand this test.